app/src/main/java/com/pr0gramm/app/ui/fragments/BenisGraphFragment.kt

```diff
@@ -0,0 +1,157 @@
+    private val userService: UserService by instance()
+    internal lateinit var data: BenisGraphDataSource
+    internal var firstOfDays: MutableMap<String, Long> = mutableMapOf()
+    private val dateFormat = SimpleDateFormat("dd.MM.yy")
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+
+    }
+
+    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
+        return inflater.inflate(R.layout.fragment_benis_graph, container, false)
+    }
+
+    override fun onViewCreated(view: View?, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+
+        data = BenisGraphDataSource()
+        val serie = BenisSeries(data,getString(R.string.benis_graph_title))
+        val formatter = LineAndPointFormatter(this.context,R.xml.benis_plot_line_point_formatter)
+        formatter.pointLabeler = PointLabeler<XYSeries> { p0, p1 -> if (p1 % 5 == 0) p0.getY(p1).toString() else "" }
+        plot.addSeries(serie, formatter)
+
+        PanZoom.attach(plot,PanZoom.Pan.HORIZONTAL,PanZoom.Zoom.STRETCH_HORIZONTAL)
+
+        plot.rangeStepModel = StepModel(StepMode.INCREMENT_BY_FIT, 1.0)
+        plot.backgroundPaint.color = Color.TRANSPARENT
+        plot.setBackgroundColor(Color.TRANSPARENT)
+        plot.graph.backgroundPaint.color = Color.TRANSPARENT
+        plot.graph.gridBackgroundPaint.color = Color.TRANSPARENT
+
+        plot.graph.domainSubGridLinePaint.color = Color.TRANSPARENT
+        plot.graph.linesPerDomainLabel
+        plot.linesPerDomainLabel = 2
+        plot.linesPerRangeLabel = 2
+        plot.legend.isVisible = false
+        plot.setBorderStyle(Plot.BorderStyle.NONE, null, null)
+        plot.graph.getLineLabelStyle(XYGraphWidget.Edge.LEFT)
+                .format = DecimalFormat("0")
+
+        plot.graph.getLineLabelStyle(XYGraphWidget.Edge.BOTTOM)
+                .format = object : Format() {
+
+            private val timeFormat = SimpleDateFormat("HH:mm")
+            override fun format(obj: Any, toAppendTo: StringBuffer, pos: FieldPosition): StringBuffer {
+                val ts = Math.round((obj as Number).toDouble())
+                val date = Date(ts)
+                if (firstOfDays.containsValue(ts)){
+                    return dateFormat.format(date,toAppendTo,pos)
+                }
+                return timeFormat.format(date, toAppendTo, pos)
+            }
+
+            override fun parseObject(source: String, pos: ParsePosition): Any? {
+                return null
+            }
+        }
+    }
+
+    internal inner class BenisGraphDataSource {
+
+        private var data: MutableList<BenisRecord> = mutableListOf()
+        private var minX: Long = Long.MAX_VALUE
+        private var maxX: Long = Long.MIN_VALUE
+        private var minY: Int = 0
+        private var maxY: Int = 0
+
+        init {
+            userService.loadBenis().subscribe{
```

Comment:
jetzt hast du gar kein Threading mehr. Hier fehlt nun noch ein `.subscribeOn(BackgroundScheduler).observeOn(mainThread())` und am besten noch `.compose(bindToLifecycle())` - und dann setzt du das Ding in onResume jedesmal neu.